### ISSUE TITLE
Delete with no revisions present throws

### ISSUE THREAD

**Issue by gmadkat** [javers/javers#461]
I am using JaVers in my Spring application, and when I am trying to delete an entity which has no revision history, I am seeing this exception: JaversException CANT_DELETE_OBJECT_NOT_FOUND: failed to delete object <my object>, it doesn't exists in JaversRepository
I looked at your code and noticed that is is intentional, do you have a workaround or would you consider allowing a way for us to delete entities without revisions? I am able to do a PUT (update) on an entity without revision history though, just not a delete.

Code I think this is happening in:
 ```
 public Commit createTerminalByGlobalId(String author, Map<String, String> properties, GlobalId removedId){
        Validate.argumentsAreNotNull(author, properties, removedId);
        Optional<CdoSnapshot> previousSnapshot = javersRepository.getLatest(removedId);
        if (previousSnapshot.isEmpty()){
            throw new JaversException(JaversExceptionCode.CANT_DELETE_OBJECT_NOT_FOUND,removedId.value());
        }
        CommitMetadata commitMetadata = nextCommit(author, properties);
        CdoSnapshot terminalSnapshot = snapshotFactory.createTerminal(removedId, previousSnapshot.get(), commitMetadata);
        Diff diff = diffFactory.singleTerminal(removedId, commitMetadata);
        return new Commit(commitMetadata, Lists.asList(terminalSnapshot), diff);
    }
```

We do need to support historical data with no revision history and delete them as needed so please help us do this. 

Thank you

**Comment by gmadkat** [comment]
I am happy to make any changes and submit a pull request if you are ok accepting them.

**Comment by bartoszwalacik** [comment]
And these deletes of non existing objects are called by aspects (@JaversSpringDataAuditable) ?

**Comment by gmadkat** [comment]
Yes. We have a Postgres database of legacy data with the JpaRepository objects annotated with @JaversSpringDataAuditable, and we were trying to delete them.

**Comment by bartoszwalacik** [comment]
Ok, in this case it makes sense to swallow and log the exception. We will accept a PR, please implement this catch in OnDeleteAuditChangeHandler (in aspects impl).

**Comment by gmadkat** [comment]
In order to fix,  I am trying to build JaVers using gradlew build on Mingw Windows and got an error:

```
org.javers.core.json.JsonConverterUtilTypesTest > should convert class java.io.File (\tmp\file.txt) to and from JSON FAILED
    Condition not satisfied:

    javers.jsonConverter.toJson( givenValue ) == expectedJson
    |      |             |       |            |  |
    |      |             |       \tmp\file.txt|  "/tmp/file.txt"
    |      |             "\\tmp\\file.txt"    false
    |      |                                  4 differences (76% similarity)
    |      |                                  "(\\\\)tmp(\\\\)file.txt"
    |      |                                  "(/~-~)tmp(/~-~)file.txt"
    |      org.javers.core.json.JsonConverter@696d6bb8
    org.javers.core.JaversCore@41548f5
        at org.javers.core.json.JsonConverterUtilTypesTest.should convert #expectedType (#givenValue) to and from JSON(JsonConverterUtilTypesTest.groovy:19)

821 tests completed, 1 failed
:javers-core:test FAILED

FAILURE: Build failed with an exception.

```

**Comment by bartoszwalacik** [comment]
what is Mingw Windows? do you use gradle normally or is it your first time?

**Comment by bartoszwalacik** [comment]
I'm not a Windows user but I think you don't need Mingw to run Gradle (Gradle is a Java app) simply run gradlew.bat

**Comment by gmadkat** [comment]
I am seeing errors on building on Windows 10 regardless of Mingw or a command prompt. I can attach a log. I went to Mingw in the hope that a Linux'y environment would help. I can attach a log. I can build on a Mac too and will try that, but our corporate environment is windows.

**Comment by mwesolowski** [comment]
@gmadkat You're right. I've reproduced this error. Currently one of our tests assumes that it is executed in UNIX environment and fails in Windows. We will fix this.

**Comment by bartoszwalacik** [comment]
@gmadkat, try latest master, @mwesolowski has just pushed a fix for Windows

**Comment by gmadkat** [comment]
I just tried it, I can build on Windows 10 now, thanks so much! 
Gowri

**Comment by gmadkat** [comment]
I am implementing swallowing the exception in OnDeleteAuditChangeHandler  and I am running into one problem. The throwing of the JaversException in the code below is setting the rollback flag and forcing my transaction to not commit and I cannot recover from this. Please advise on how to implement this so that a lack of history does not set the rollback flag in my transaction?

```
public Commit createTerminalByGlobalId(String author, Map<String, String> properties, GlobalId removedId){
        Validate.argumentsAreNotNull(author, properties, removedId);
        Optional<CdoSnapshot> previousSnapshot = javersRepository.getLatest(removedId);
        if (previousSnapshot.isEmpty()){
            throw new JaversException(JaversExceptionCode.CANT_DELETE_OBJECT_NOT_FOUND,removedId.value());
        }
```

**Comment by bartoszwalacik** [comment]
This is a good idea. So instead of swallowing the exception we will chec if given object exists and call `javers.commitShallowDelete()` only if given object exists

**Comment by gmadkat** [comment]
Thank you, I tried to do this and some tests broke, is this something you would be more comfortable fixing than me?

**Comment by bartoszwalacik** [comment]
ok, I will take a look

**Comment by gmadkat** [comment]
I actually tried the below and it appeared to fix the bug, but I do not like returning null, and i noticed that the returned commit is not used anyways in OnDeleteAuditChangeHandler 

Please let me know if the below patch is acceptable, I will clean out the commented code if so and submit. 

```
$ git diff
diff --git a/javers-core/src/main/java/org/javers/core/JaversCore.java b/javers-core/src/main/java/org/javers/core/JaversCore.java
index 6b2b3d9..97715ee 100644
--- a/javers-core/src/main/java/org/javers/core/JaversCore.java
+++ b/javers-core/src/main/java/org/javers/core/JaversCore.java
@@ -92,7 +92,7 @@ class JaversCore implements Javers {
         argumentsAreNotNull(author, properties, deleted);

         Commit commit = commitFactory.createTerminal(author, properties, deleted);
-
+        if (commit == null) return null;
         repository.persist(commit);
         logger.info(commit.toString());
         return commit;
@@ -109,6 +109,7 @@ class JaversCore implements Javers {
         argumentsAreNotNull(author, properties, globalId);

         Commit commit = commitFactory.createTerminalByGlobalId(author, properties, globalIdFactory.createFromDto(globalId));
+        if (commit == null) return null;

         repository.persist(commit);
         logger.info(commit.toString());
diff --git a/javers-core/src/main/java/org/javers/core/commit/CommitFactory.java b/javers-core/src/main/java/org/javers/core/commit/CommitFactory.java
index 77a4116..3122c2d 100644
--- a/javers-core/src/main/java/org/javers/core/commit/CommitFactory.java
+++ b/javers-core/src/main/java/org/javers/core/commit/CommitFactory.java
@@ -47,7 +47,7 @@ public class CommitFactory {
         Validate.argumentsAreNotNull(author, properties, removedId);
         Optional<CdoSnapshot> previousSnapshot = javersRepository.getLatest(removedId);
         if (previousSnapshot.isEmpty()){
-            throw new JaversException(JaversExceptionCode.CANT_DELETE_OBJECT_NOT_FOUND,removedId.value());
+           return null; // throw new JaversException(JaversExceptionCode.CANT_DELETE_OBJECT_NOT_FOUND,removedId.value());
         }
         CommitMetadata commitMetadata = nextCommit(author, properties);
         CdoSnapshot terminalSnapshot = snapshotFactory.createTerminal(removedId, previousSnapshot.get(), commitMetadata);
diff --git a/javers-core/src/test/groovy/org/javers/core/JaversCommitE2ETest.groovy b/javers-core/src/test/groovy/org/javers/core/JaversCommitE2ETest.groovy
index 664d34e..8f9e0ad 100644
--- a/javers-core/src/test/groovy/org/javers/core/JaversCommitE2ETest.groovy
+++ b/javers-core/src/test/groovy/org/javers/core/JaversCommitE2ETest.groovy
@@ -94,7 +94,7 @@ class JaversCommitE2ETest extends Specification {
                           { j -> j.commitShallowDeleteById("some.login", instanceId(1,SnapshotEntity))} ]
         opType << ["using object instance","using globalId"]
     }
-
+/*
     def "should fail when deleting non existing object"() {
         given:
         def javers = javers().build()
@@ -107,7 +107,7 @@ class JaversCommitE2ETest extends Specification {
         JaversException exception = thrown()
         exception.code == JaversExceptionCode.CANT_DELETE_OBJECT_NOT_FOUND
     }
-
+*/
     def "should create initial commit for new objects"() {
         given:
         def javers = javers().build()

```

**Comment by gmadkat** [comment]
I just wanted to follow up on this issue and see whether the above makes sense and is a fix, or if you are fixing it, can you tell me a timeframe for this? We need it asap for our release. Thank you!

**Comment by bartoszwalacik** [comment]
We are working on pull requests

**Comment by gmadkat** [comment]
I can do that, does my diff above look like a reasonable fix? I can submit it in a PR

**Comment by bartoszwalacik** [comment]
Well, i dont know. It's not human readable

**Comment by gmadkat** [comment]
@bartoszwalacik  Ok let me do a pull request and can go from there.

**Comment by gmadkat** [comment]
@bartoszwalacik  I submitted a pull request for this fix, please let me know if it looks ok

**Comment by bartoszwalacik** [comment]
see https://github.com/javers/javers/pull/477/files

**Comment by gmadkat** [comment]
Thank you, when will this be merged?

**Comment by bartoszwalacik** [comment]
merged, it will be released today

**Comment by bartoszwalacik** [comment]
released in 2.8.0
